compio-driver/src/asyncify.rs

```diff
@@ -0,0 +1,47 @@
+use std::thread::JoinHandle;
+
+use crossbeam_channel::{bounded, Receiver, Sender, TrySendError};
+
+type BoxClosure = Box<dyn FnOnce() + Send>;
+
+fn worker(receiver: Receiver<BoxClosure>) -> impl FnOnce() {
+    move || {
+        while let Ok(f) = receiver.recv() {
+            f();
+        }
+    }
+}
+
+pub struct AsyncifyPool {
+    sender: Sender<BoxClosure>,
+    receiver: Receiver<BoxClosure>,
+    threads: Vec<JoinHandle<()>>,
+}
+
+impl AsyncifyPool {
+    pub fn new() -> Self {
+        let (sender, receiver) = bounded(0);
+        let threads = vec![std::thread::spawn(worker(receiver.clone()))];
+        Self {
+            sender,
+            receiver,
+            threads,
+        }
+    }
+
+    pub fn dispatch(&mut self, f: impl FnOnce() + Send + 'static) {
+        match self.sender.try_send(Box::new(f) as BoxClosure) {
+            Ok(_) => {}
+            Err(e) => match e {
+                TrySendError::Full(f) => {
+                    self.threads
+                        .push(std::thread::spawn(worker(self.receiver.clone())));
```

Comment:
Add a hard limit here so there won't be too many threads running if some burst happens. Maybe making dispatch async and sleep inside if hard limit reached.

Reply:
Cannot use async fn in driver:(

But we can set a hard limit to 256 maybe.

Reply:
I just realized it's in driver. I thought this should be in runtime? Does thread pool use platform specific API's?

Reply:
The thread pool itself is cross-platform, but io-uring doesn't need it. It would be painful to move it to runtime. In that way, the high-level implementors must have knowledge of whether an operation is async or not.